packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -300,4 +375,49 @@ contract ERC20Vault is BaseVault {
             ctokenDecimal: ctoken.decimals
         });
     }
+
+    /// @dev Handles the burn/mint per native/non-native tokens
+    /// @param onToken The token contract (or adapter) the operation needs to be performed.
+    /// @param nativeBridgedTokenProxy Important on L2, when we need the preDeployed contract (e.g.:
+    /// USDC).
+    /// @param toOrFrom The address to or from.
+    /// @param amount The amount to be burnt/minted.
+    /// @param isNative Flag indicating, if this is a native L1 token or not.
+    /// @param isMint True if mint, false for burn.
+    function _executeBurnOrMint(
+        address onToken,
+        address nativeBridgedTokenProxy,
+        address toOrFrom,
+        uint256 amount,
+        bool isNative,
+        bool isMint
+    )
+        private
+    {
+        // Need an adapter: BridgedERC20 or custom (e.g.: USDC)
+        address adapter = isNative ? onToken : resolve("bridged_erc20_adapter", false);
+        // Need the token (proxy) contract
+        address tokenContract = isNative ? nativeBridgedTokenProxy : onToken;
+        if (isMint) {
+            (bool success,) = adapter.delegatecall(
+                abi.encodeWithSignature(
+                    "mint(address,address,uint256)", tokenContract, toOrFrom, amount
+                )
+            );
+
+            if (!success) {
+                revert VAULT_MINT_ERROR();
+            }
+        } else {
+            (bool success,) = adapter.delegatecall(
+                abi.encodeWithSignature(
+                    "burn(address,address,uint256)", tokenContract, toOrFrom, amount
+                )
+            );
+
+            if (!success) {
+                revert VAULT_BURN_ERROR();
+            }
```

Comment:
More code reuse when just storing `"mint(address,address,uint256)"`/`"burn(address,address,uint256)"` in a string based on `isMint`. Then all other code could be shared.